libaugrim/src/time.rs

```diff
@@ -19,7 +19,7 @@ use std::time::Duration;
 
 pub trait Time: Add<Duration, Output = Self> + PartialOrd + Copy + Debug {}
 
-pub trait TimeSource {
+pub trait TimeSource: Clone {
```

Comment:
Same constraint question

Reply:
Yes. It is probably better to specify it on TwoPhaseCommitAlgorithm because it was only added because of an implementation detail of TwoPhaseCommitAlgorithm and could probably have been avoided with Arc/Mutex (for example), so its clearly not fundamental.

Removed this commit.